### DISCUSSION TITLE
mssql anonymous block case statment Error

### DISCUSSION THREAD

**Discussion by bonohemi** [dbeaver/dbeaver#22848] [Q&A]
DBeaver 23.3.4 Community version

mssql: 
An error occurs when using the case statement in begin block.

begin
    select getdate();

    SELECT 
        CASE
            WHEN 1=1 THEN 'True'
            ELSE 'False'
        END,
        *
    FROM
        SYS.SYSOBJECTS
    WHERE 1=1
        AND name LIKE 'sys%'

end

SQL Error [102] [S0001]: Invalid syntax near 'sys%'

Perhaps it recognizes the end of the begin statement as the end of the case statement.
Contact us to see if it is possible to fix this bug.

![image](https://github.com/dbeaver/dbeaver/assets/5730263/91d482d7-b5ab-4c03-8ff1-6194d215b8a1)

**Comment by mkgrgis** [comment]
Look like this is internal problem of MSSQL sever instance. 100% this is not DBeaver problem.

**Comment by LonwoLonwo** [comment]
It will work in case you highlight the whole query and then execute this query.
